api/src/main/java/org/openmrs/api/impl/PatientServiceImpl.java

```diff
@@ -128,9 +128,9 @@ public Patient savePatient(Patient patient) throws APIException {
 			checkPatientIdentifiers(patient);
 		}
 
-		setPreferredPatientIdentifier(patient);
-		setPreferredPatientName(patient);
-		setPreferredPatientAddress(patient);
+		setPatientDetail(patient, new PatientIdentifier().getClass());
```

Comment:
i would just duplicate it by removing setPreferredPateintIdentifier to simply setPatientIdentifier, however i havent taken deep where setPreferred were used

Reply:
@sherrif10 Could you please, throw more light on this?